modules/hal_nordic/nrfs/dvfs/ld_dvfs_handler.c

```diff
@@ -139,9 +139,19 @@ static void dvfs_service_handler_scaling_background_job(enum dvfs_frequency_sett
 	}
 }
 
+/* Update MDK variable which is used by nrfx_coredep_delay_us (k_busy_wait). */
+static void dvfs_service_update_core_clock(enum dvfs_frequency_setting oppoint_freq)
```

Comment:
This would need to be called every time ld_dvfs_configure_hsfll is called so also init and dvfs_service_handler_prepare_to_scale. I would move this inside ld_dvfs_configure_hsfll function, what do You think?